### PR TITLE
Move GetCheckpointAction under the monitor namespace

### PR DESCRIPTION
**Pull request by ywangd**
In #80984, a new GetCheckpointAction is added under the namespace of
`indices:internal/`. This is a new namespace and there it is not automatically
covered by the `manage` index privilege. Since it is explicilty added to 
`view_index_metadata`, it means `view_index_metadata` is no longer a 
subset of `manage`. This is unexpected.

After discussion, instead of adding this new namespace to `manage`, we agreed
to move the new action under `monitor` and drop the new namespace. 
IIUC, the `internal` is used to indicate that this action is internal to a bigger process
and it cannot be called on its own and should be kept as an implementation detail.
However, what privilege an action should have is an orthogonal concern to how it 
should be used. The function of this action is more of `monitor` (similar to the existing
[GetGlobalCheckpoint API](https://www.elastic.co/guide/en/elasticsearch/reference/current/get-global-checkpoints.html))

This PR also adds a few tests to ensure certain subset relationships between 
privileges, e.g. "view_index_privilege" is a subset of "manage".

### CODE REVIEW

**Comment by ywangd** [comment]
I kept the explicit grant of `GetCheckpointAction` to `view_index_metadata` because I believe ML currently already requires `view_index_metadata` for relevant process and there is no intention to change that for the time being. @hendrikmuhs Please feel free to correct me. Thanks!

**Comment by ywangd** [comment]
The [test failure](https://gradle-enterprise.elastic.co/s/lponfcfjlsktc/tests/:x-pack:plugin:ml:internalClusterTest/org.elasticsearch.xpack.ml.integration.DatafeedCcsIT/testDatafeedWithCcsRemoteHealthy?anchor=eyJvdXRwdXRMaW5lIjoiMC05OTkifQ&focused-execution=1&top-execution=1#Lundefined) is unrelated. It does not even have security enabled.

**Comment by hendrikmuhs** [comment]
> I kept the explicit grant of `GetCheckpointAction` to `view_index_metadata` because I believe ML currently already requires `view_index_metadata` for relevant process and there is no intention to change that for the time being. @hendrikmuhs Please feel free to correct me. Thanks!

Yes, that's correct.

**Review by hendrikmuhs — APPROVED** [review]
LGTM

However, I am asking you to add documentation to the code, e.g. in `Constants.java` you could add a code doc header explaining the different namespaces and purposes. Although this is test code, every new action has to be added here, so a developer would automatically stumble upon this documentation (or you find a better place and link it, whatever). Your explanations in the PR are great, but will be lost.

**Review by tvernum — APPROVED** [review]

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/IndexPrivilege.java [hunk 1]
```diff
@@ -52,6 +52,19 @@
...
+ * first level comprises `admin`, `monitor`, `data` which generally categorize an action into either an admin
+ * related function, or a monitoring related function or a user-data related function. Subsequent levels further
+ * narrow down the category until the meaning is specific enough.
+ *
+ * Note that these terms are meant to categorize what the action does, *not* how it should be invoked. This means
+ * whether an action is accessible via REST API should not contribute to its naming.
+ *
+ * Also note that the `internal:transport/proxy/` prefix is automatically added and stripped for actions that go
+ * through a CCR/CCS proxy. No action should be explicitly named like that.
+ */
```
Comment (ywangd) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/IndexPrivilege.java:67]: > However, I am asking you to add documentation to the code, e.g. in Constants.java you could add a code doc header explaining the different namespaces and purposes. Although this is test code, every new action has to be added here, so a developer would automatically stumble upon this documentation (or you find a better place and link it, whatever). Your explanations in the PR are great, but will be lost.

@hendrikmuhs I added this comment to `IndexPrivilege`. I hope this read right to you and potentially be useful in future. Thanks!

**Pull request merged**